activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -27,6 +27,8 @@ def sql(event)
 
       return if IGNORE_PAYLOAD_NAMES.include?(payload[:name])
 
+      pool_name = payload[:connection]&.pool&.pool_config&.db_config&.name
```

Comment:
Connection management in Rails is really complex and this is conflating 2 ideas into one that we've been working to separate.

`pool_name` is the owner class name, and is either ActiveRecord::Base or another parent class that connects to the database, like AnimalsBase.

`db_config.name` is the key in the database.yml for the connection. I don't think we should include the db_config.name key in the logs and instead should add the parent class name.

To accomplish this you can get that form the `pool_config` like this `payload[:connection]&.pool.pool_config.connection_specification_name`. You don't need to check for nils all the way down because you cannot have a pool config without a connection_specification_name.

If you want to know whether it's a writing or reading connection then we should print out the class name with the handler role (writing/reading/etc) rather than the configuration name.

Reply:
The parent class names can already be identified from the log like `Dog Load (0.1ms) ...` in popular cases. Adding the connection name is more useful for replicas or horizontal sharding use cases. For example, in applications which determine target shards based on requests, it is useful to know which shard is actually used for queries.

How do you feel to adding `current_pool_key` and `prevent_writes` state to the log?

Reply:
`current_pool_key` is an internal concern and isn't going to give the user much information. It's always default unless you're using shards and doesn't differ across the writing/reading connections.

Reply:
What I want to provide more information about is the sharding use case. If `current_pool_key ` is default, we can omit the shard name from the log.

I think logging based on `prevent_writes` state will provide more information on writing/reading connection use case too.

Reply:
What I want to provide more information about is the sharding use case. If current_pool_key is default, we can omit the shard name from the log.

I think logging based on prevent_writes state will provide more information on writing/reading connection use case too.

Reply:
@eileencodes  I have opened pull request https://github.com/rails/rails/pull/39720 that uses pool_key for debug logging and record's shard tracking. I'd appriciate if you could check this one. Thanks!